tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc

```diff
@@ -477,16 +492,76 @@ void ral_qgemm_acl_s8_s8_s8_per_channel(
                 << "\n";
   }
 }
+#endif  // TAO_AARCH64
+
+#if defined(TAO_X86)
+// The data format:
+// input: s8 per-tensor, weight: s8 per-channel, result: s8 per-tensor
+// is hard-coded.
+// TODO(DISC): support more quantization dtype & scheme
+void ral_qgemm_onednn_s8_s8_s8_per_channel(
+    ExecutionContext* ctx, opaque_t /*stream_handle*/,
+    MemRefType<int8_t, 2> input, MemRefType<int8_t, 2> weight,
+    MemRefType<float, 0> inputScales, MemRefType<int32_t, 0> inputZeroPoints,
+    MemRefType<float, 1> weightScales, MemRefType<int32_t, 1> weightZeroPoints,
+    MemRefType<float, 0> resultScales, MemRefType<int32_t, 0> resultZeroPoints,
+    MemRefType<int8_t, 2> result, bool tp_a, bool tp_b, bool weight_is_const) {
+  CpuTimer timer("ral_qgemm_onednn_s8_s8_s8_per_channel");
+  if (isEmptyMemref(input) || isEmptyMemref(weight) || isEmptyMemref(result)) {
+    TAO_VLOG(1) << "ral_qgemm_onednn_s8_s8_s8_per_channel: early return for "
+                   "empty tensor";
+    return;
+  }
+  int64_t m = tp_a ? input.sizes[1] : input.sizes[0];
+  int64_t k = tp_a ? input.sizes[0] : input.sizes[1];
+  if (k != (tp_b ? weight.sizes[1] : weight.sizes[0])) {
+    ctx->signalError(Context::FAILURE,
+                     "mismatch contraction dim for quantization gemm");
+    return;
+  }
+  int64_t n = (tp_b ? weight.sizes[0] : weight.sizes[1]);
+
+  ideep::tensor input_t{dims{m, k}, ideep::data_type::s8,
+                        tp_a ? format_tag::ba : format_tag::ab, input.data};
+  ideep::tensor weight_t{dims{k, n}, ideep::data_type::s8,
+                         tp_b ? format_tag::ba : format_tag::ab, weight.data};
+  ideep::tensor output_t{dims{m, n}, ideep::data_type::s8, format_tag::ab,
+                         result.data};
+
+  std::vector<float> input_scales({inputScales.data[0]});
```

Comment:
Is it safe to ignore zero_points input here?

Reply:
I attach the zero_point of input/weight/output to corresponding tensor